### DISCUSSION TITLE
Request that rapids object file "A white water" will no longer block or delay undocking of boat rides at boat ride undocking and exit

### DISCUSSION THREAD

**Discussion by jamtraks** [OpenRCT2/OpenRCT2#18521] [Feature requests]
### Operating System

Windows 11 64-bit

### OpenRCT2 build

v0.4.0, c6302a1 on master

### Describe the issue

I have built a few parks with boat rides, and the rapids  object file "A white water" in the water where the boats roam, are blocking the boats from undocking at the exit. The guests can't get off of the boat rides and are complaining that they want to get off of the ride.
I want the rapids to be shown in the water but i also want the boats to be able to undock.
I think this variable should be completely removed from the game. If not, it should be a feature that can be turned on or off, the fact that the rapids block the boats from undocking.
Over all i think this variable  is silly, and in real life this wouldn't have anything to do with boats undocking or not at an amusement park.
Would this ruin any park or would anyone miss the blocking of boat undocking due to rapids?
Is it removable from the game?

### Steps to reproduce

1.place rapids object file "A white water" in the water where boats roam and near undocking station and observe  that the boats then do not undock to let off at the ride station exit.
2. 
3. 

### Attachments
"C:\Users\jamtr\OneDrive\Documents\OpenRCT2\save\Crazy Castle By Josh Marsilio 4.17.sv6"
_No response_

**Comment by ocalhoun6** [comment]
Are you talking about "rapids" as in a custom scenery object, or "rapids" as in a section of rapids track from the river rapids ride?

(For an interim workaround in either case, you could place extra track on your boat hire ride before the station platform -- free roaming boats will dock with that track instead of the end of the platform, and you can place the boat hire track so that it leads to an open space where boats are able to go. You can then make the track invisible using the tile inspector, so it looks like boat hire boats smoothly go over the rapids and go to their station.)

**Comment by jamtraks** [comment]
I am referring to free roam boats such as rowing boats and bumper boats. I have a few set ups in a few of my parks, with no lead-track, they free roam fine with no rapid scenery in the water. but If i place rapids object scenery in the water where the boats free-roam and near the undocking exit, the boats don't lead into the exit to undock. The rapid object file itself prevents the boats from undocking. I've tested the boat rides with and  without the rapid object in the water, and  with the rapid object in the water the boats don't undock, and  without it they do. I was told by another person here that the rapid object does in fact prevent the boats from undocking.
I don't want the boats to have a track, i want them to be free-roam. They undock only if there is NO rapid object file in the water.

**Comment by duncanspumpkin** [comment]
To me it sounds like you want a flag that custom scenery can set that means boats ignore it when pathing. You would need to of course modify your custom object to add the flag but I don't see any major harm in doing this.

**Comment by jamtraks** [comment]
Yes, i want to be able to null the blocking of boats by the object file "A white water". Or i think the blocking of boat redocking by that object file should be permanently removed from the game somehow.
How would i do this myself?

**Comment by duncanspumpkin** [comment]
Could you provide the object in question

**Comment by jamtraks** [comment]
"A white water"

**Comment by jamtraks** [comment]
Maybe there is a different white-water-rapids object file (named something else other than "a white water") that is visually identical to "a white water" but does not block the redocking of free roam boats, and i am only using the wrong rapids object file. 
I don't know if there is or isn't though I haven't searched for or checked for any other white-water-rapids object files.

**Comment by spacek531** [comment]
![image](https://user-images.githubusercontent.com/3454156/200191707-d8995f4b-3f4a-4f67-8527-d3b9fba039bb.png)
Is this your setup?

**Reply by jamtraks** [reply]
No, i actually have nothing set up like that there where the rapid is literally at the entrance or exit. I made a few free roam boat rides though and the rapids being where the boats float during the ride blocks them from redocking. but only in one case where i have free roaming rowing boats.
![image](https://user-images.githubusercontent.com/17103908/200192088-00860beb-5018-4e2c-82ec-c221020e1306.png)
i am a beginner player and i don't know how to make pretty rapids like they do at nedesigns, but still i am wondering now is there a rapids object that might not block or influence the boats at all.
Sorry if i'm making a big deal out of nothing. Feel free to re close this topic if it has no relevance.

**Comment by jamtraks** [comment]
Would messing with the collision detection boxes in the tile inspector menu after clicking on a tile where there is an "a white water" river rapid object, change how the rapids do or do not block the boats floating near or across them or not? What is collision detection for "a white water"? what does checking or unchecking the collision detection boxes in the tile inspector menu do? when and how is it used?
Thanks...
